src/include/function/cast/cast_functions.h

```diff
@@ -87,27 +88,81 @@ inline std::string CastToString::castToStringWithVector(
 
 template<typename SRC, typename DST>
 static inline void numericDownCast(SRC& input, DST& result, const std::string& dstTypeStr) {
-    if (input < std::numeric_limits<DST>::min() || input > std::numeric_limits<DST>::max()) {
+    if ((signed)input < std::numeric_limits<DST>::min() ||
```

Comment:
(signed)input looks incorrect to me.
numericDownCast(uint64_t, int64_t).
For example, the input is uint64t_max. What will the result be for (signed)(uint64t_max)?

Reply:
I'll revert all the changes related to unsigned cast to signed. And will only keep those changes that make the tests pass.